editor/filesystem_dock.cpp

```diff
@@ -196,24 +195,28 @@ Vector<String> FileSystemDock::_compute_uncollapsed_paths() {
 			uncollapsed_paths.push_back(favorites_item->get_metadata(0));
 		}
 
-		TreeItem *resTree = root->get_first_child()->get_next();
-		if (resTree) {
-			Vector<TreeItem *> needs_check;
-			needs_check.push_back(resTree);
+		// BFS to find all uncollapsed paths of the resource directory.
+		TreeItem *res_subtree = root->get_first_child()->get_next();
+		if (res_subtree) {
+			List<TreeItem *> queue;
+			queue.push_back(res_subtree);
 
-			while (needs_check.size()) {
-				if (!needs_check[0]->is_collapsed()) {
-					uncollapsed_paths.push_back(needs_check[0]->get_metadata(0));
-					TreeItem *child = needs_check[0]->get_first_child();
-					while (child) {
-						needs_check.push_back(child);
-						child = child->get_next();
+			while (queue.size()) {
+				TreeItem *ti = queue.back()->get();
+				queue.pop_back();
+				if (!ti->is_collapsed() && ti->get_child_count() > 0) {
+					Variant path = ti->get_metadata(0);
+					if (path) {
+						uncollapsed_paths.push_back(path);
 					}
 				}
-				needs_check.remove_at(0);
+				for (int i = 0; i < ti->get_child_count(); i++) {
+					queue.push_back(ti->get_child(i));
+				}
 			}
 		}
 	}
+
```

Comment:
Such blank lines are unnecessary, because the lone `}` already acts as separator.